src/board.cpp

```diff
@@ -0,0 +1,179 @@
+#include <iostream>
+#include "board.h"
+#include "cpu.h"
+
+
+Board::Board(size_t board_size) : players_(), num_humans_(0), board_size_(board_size) {}
+
+void Board::init(size_t new_board_size) {
+  // If board_size is 0, keep the current size
+  if (new_board_size > 0) {
+    board_size_ = new_board_size;
+  }
+  for (unsigned int i = 0; i < get_num_players(); ++i) {
+    // Player ids are 1-indexed
+    players_[i]->init(i + 1, board_size_);
+  }
+}
+
+void Board::reset() {
+  players_.clear();
+  num_humans_ = 0;
+}
+
+void Board::print_board() const {
+  GUI::print_header("NUMBERS ROYALE");
+  // Print players in reverse, so player 1 is on the bottom
+  for (int i = get_num_players() - 1; i >= 0; --i) {
+    std::string playerText = players_[i]->name() + ": " +
+        std::to_string(players_[i]->get_score()) + " points";
+    GUI::print_item(playerText.c_str());
+    GUI::print_item("Moves:");
+    GUI::print_item(players_[i]->get_moves_string().c_str(), true);
+  }
+  GUI::print_border();
+}
+
+void Board::print_scores() const {
+  int winner_id = check_game_winner();
+  switch (winner_id) {
+    case TIE_GAME:
+      GUI::print_header("TIE GAME!");
+      break;
+    case CONTINUE_GAME:
+      GUI::print_header("SCOREBOARD", false);
+      break;
+    default:
+      std::string winner_text = players_[winner_id - 1]->name() + " WINS!";
+      GUI::print_header(winner_text.c_str());
+      break;
+  }
+  // Sort players by descending score
+  std::vector<Player*> sorted_player_view;
```

Comment:
```suggestion
  std::vector<const Player&> sorted_player_view;
```
There's no need for this to be a pointer right? const regardless

Reply:
```suggestion
  std::vector<Player*> sorted_player_view(players().size());
```
there's a significant performance bump if you pre-initialize the size if it's known ;)